### PR TITLE
[Not for Merge] Drop TwoIfBySeaEndTurnDelegate

### PR DESCRIPTION
**Pull request by DanVanAtta**
Per deprecation comments, was waiting for incompatible release to drop this code.

 

## Functional Changes

[ ] New map or map update
[ ] New Feature
[ ] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[x] Covered by existing automated tests
[ ] Covered by newly added automated tests
[x] Manually tested
[ ] No testing done

- Launched a local game.
- Looked for TwoIfBySea in map repo, not found.

### CODE REVIEW

**Comment by Cernelius** [comment]
The current prerelease is not map incompatible with the current stable.

**Comment by Cernelius** [comment]
This is what @RoiEXLab said:
https://forums.triplea-game.org/topic/595/short-attachment-names/17
By "will be removed in a future version" I just wanted to strongly indicate the twoIfBySea delegates shouldn't be used to enable us to either rename it in an incompatible release, or to incorporate them into the existing delegates and to remove them from the twoIfBySea package afterwards.

File: game-core/src/main/java/games/strategy/engine/data/gameparser/XmlGameElementMapper.java [hunk 1]
```diff
@@ -101,23 +101,10 @@ public XmlGameElementMapper(
         .put("TechActivationDelegate", TechActivationDelegate::new)
         .put("TechnologyDelegate", TechnologyDelegate::new)
         .put("UserActionDelegate", UserActionDelegate::new)
-        .putAll(newTwoIfBySeaDelegateFactories())
         .putAll(auxiliaryDelegateFactoriesByTypeName)
         .build();
   }
 
-  @SuppressWarnings(
-      "deprecation") // required for map compatibility; remove upon next map-incompatible release
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/engine/data/gameparser/XmlGameElementMapper.java]: @RoiEXLab would you mind clarifying this comment? What exactly can be removed? The delegate 'TwoIfBySeaEndTurnDelegate" is used in `class_variations/../four_if_by_sea.xml`

**Comment by DanVanAtta** [comment]
@Cernelius can this PR be resurrected? https://github.com/triplea-maps/classic_variations/pull/3/files

If we made that update, it looks like the delegate code being removed here would become unused. The comment quoted: https://github.com/triplea-game/triplea/pull/5242#issuecomment-532683765 from @RoiEXLab is pretty confusing.

@RoiEXLab, can you clarify the state of the twoIfBySea end turn delegate, is it slated to be removed? Is there a deprecation path?

**Comment by RoiEXLab** [comment]
@DanVanAtta here's the story I recall.
I believe I asked the map makers some time ago if they knew what made the twoIfBySea delegates special. IIRC no one really seemed to know, there were only a couple maps that actually used them.
So ideally those few maps were migrated to use the other delegates, so the twoIfBySea delegates can be removed from the code

**Comment by DanVanAtta** [comment]
Okay, sounds like the update: https://github.com/triplea-maps/classic_variations/pull/3/files @Cernelius proposed would do that.

@Cernelius do you know if that closed PR is still valid, can we re-open it?

**Comment by Cernelius** [comment]
Unless I'm missing anything, what that PR would still does is merely fully substituting the deprecated and redundant "twoIfBySea" delegate with the regular one, that I understand it is already recalling anyways. So that is no more no less valid that doing the same thing again, on a new PR. You can merge that PR if you would be doing the same thing yourself.

The problem with this, is that, apparently, the games using the "twoIfBySea" appear to be supposed to work differently than they do. I suppose this delegate was, therefore, doing something else, in the past. Currently, all those maps are substantially bugged, as they recall something that is not really anymore supported, but I'm not even sure of this. The only problem with substituting the "twoIfBySea" delegate with the regular one is hiding this matter, so that you will not know anymore what are those maps, if you'll even figure out what that delegate is supposed to do. I'm not sure converting those games like that is really better than deleting them (and deleting them may be a good choice, if they are based on something that is not anymore supported).

With a huge warning here I'm really guessing, I think that the "twoIfBySea" was a delegate for allowing cumulative placement abilities per territory. That is, if having 1 factory in a territory allows to place X, having N factories allows to place NX.

Finally, I did that only for that game, but I understand there are a few others currently using the "twoIfBySea" delegate. I don't think I care at all about (or would ever play) any of them, personally.

**Comment by Cernelius** [comment]
Basically, if anyone want to bother updating all those maps to use the regular delegate, instead of the "twoIfBySea" one, it would be better to track them all in a Wiki, as likely bugged (partially unsupported) games, with a mention that they used to work with that delegate ("twoIfBySea"). Or maybe noting it somewhere in their repositories (where?)?

I don't think it makes sense to merge that PR of mine without converting all other games having "twoIfBySea" too. That PR was intended to be the first one of a number of them, covering all cases. I don't even know why I did it at all, since I don't care about any of those games; I suppose I just wanted to help @RoiEXLab.

**Comment by Cernelius** [comment]
References:

This is the original main PR touching this matter:
https://github.com/triplea-game/triplea/pull/3367

Here I asked about this matter again, and nobody gave me an actual answer on the "twoIfBySea" item:
https://github.com/triplea-game/triplea/issues/3462

Here is a bug report that I opened more than 4 years ago, about two maps, that is apparently related to the "twoIfBySea" delegate (I don't like or play that map; I was just spectating it):
https://sourceforge.net/p/triplea/bugs/1102/

As I said, I'm really dubious whether it would be a better move to just delete all those games (as well as deleting the maps having only those games), instead of updating them. And I think they should be tracked as probably bugged if updated, but I don't know how.

**Comment by DanVanAtta** [comment]
Appreciate the background info on this @Cernelius ; @RoiEXLab I think you're the only one that has the background context to solve the TODO comment without investigating for a few hours the full extent of the background, I'll leave it to you to solve the TODO here and drop the delegate however would be appropriate.

**Comment by RoiEXLab** [comment]
@DanVanAtta Sorry to disappoint, but I really don't have a clue on how to tackle this task.
So I'm really relying on map makers as well.

**Pull request closed without merging**